File: IT-Auto-Response/it_response.py
import datetime

def generate_fcm_number(date):
    """Generate FCM number based on today's date."""
    return f"FCM{date.strftime('%m%d%y')}"

def password_reset_case(expiry_date, today):
    """Create description and resolution for password reset case."""
    description = f"The client is requesting a password reset because they are not a student since {expiry_date} and are requesting to change the password."
    fcm_number = generate_fcm_number(today)
    resolution = f"{fcm_number}: Verified client information. Added phone number and personal email as MyPassword sign-in methods. Provide the MyPassword information to change the password under 'Forgot My Password' > 'Reset Password'."
    return description, resolution

def disabled_account_case(today):
    """Create description and resolution for disabled account case."""
    last_semester = input("When was the last semester the client attended STC? (e.g., Fall 2020, Spring 2023): ")
    description = f"The client is requesting support because they have a disabled account. The client is not a student since 2022 but has a current application for this semester. The last semester the client attended was {last_semester}."
    fcm_number = generate_fcm_number(today)
    resolution = f"{fcm_number}: Verified client information. Re-enabled account. Assisted client in resetting password online. Verified client could sign in. Registered the two sign-in methods on the MyPassword Assistant."
    return description, resolution

def password_reset_email_case(today):
    """Create description and resolution for password reset email case."""
    email_date = input("When was the email sent? (MM/DD/YY): ")
    email_time = input("At what time was the email sent? (HH:MM): ")
    description = f"Client sent an email on {email_date} at {email_time} requesting support with a password reset."
    fcm_number = generate_fcm_number(today)
    resolution = f"{fcm_number}: Client received password reset auto-response and our contact information."
    return description, resolution

def main():
    # First run asks for today's date
    today_str = input("Enter today's date (MM/DD/YY): ")
    today = datetime.datetime.strptime(today_str, "%m/%d/%y")
    
    while True:
        # Ask for the type of ticket
        print("\nPlease select the type of ticket:")
        print("1. Password Reset")
        print("2. Disabled Account")
        print("3. Compromised Account")
        print("4. Password Reset Email")
        ticket_type = input("Enter the number corresponding to the ticket type: ")
        
        if ticket_type == '1':
            expiry_date = input("When did the account expire? (MM/DD/YY): ")
            description, resolution = password_reset_case(expiry_date, today)
        elif ticket_type == '2':
            description, resolution = disabled_account_case(today)
        elif ticket_type == '4':
            description, resolution = password_reset_email_case(today)
        else:
            print("This ticket type is not yet implemented.")
            continue
        
        print("\nDescription:\n", description)
        print("\nResolution:\n", resolution)
        
        # Ask if they want to continue
        continue_request = input("Do you want to handle another request? (yes/no): ").lower()
        if continue_request != 'yes':
            break

if __name__ == "__main__":
    main()

File: IT-Auto-Response/it_response.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const moment = require('moment');

function generateFcmNumber(date) {
    // Generate FCM number based on today's date.
    return `FCM${moment(date).format('MMDDYY')}`;
}

function passwordResetCase(expiryDate, today) {
    // Create description and resolution for password reset case.
    const description = `The client is requesting a password reset because the password expired since ${expiryDate} and is requesting to change it.`;
    const fcmNumber = generateFcmNumber(today);
    const resolution = `${fcmNumber}: Verified client information. Added phone number and personal email as MyPassword sign-in methods. Provide the MyPassword information to change the password under 'Forgot My Password' > 'Reset Password'.`;
    return { description, resolution };
}

function authenticationErrorCase(today) {
    // Create description and resolution for authentication error case.
    const description = `The client called to report that they are encountering an 'CAS Validation Failed' error when attempting to sign in to Jagnet.`;
    const fcmNumber = generateFcmNumber(today);
    const resolution = `${fcmNumber}: Verified client information. Informed client of username format for signing into Jagnet.`;
    const internalNotes = `Common reasons a client may receive an 'CAS Validation Failed' error include: Entering their STC email address in the 'username' field, Double-clicking the 'Enter' button to sign in, Browser cache needs to be cleared. If the client's password is not expired and you confirmed they are entering the correct information, advise them to clear the cache and cookies from their browser or try using a different browser or device.`;
    return { description, resolution, internalNotes };
}

function disabledAccountCase(today) {
    // Create description and resolution for disabled account case.
    return new Promise(resolve => {
        rl.question("When was the last semester the client attended STC? (e.g., Fall 2020, Spring 2023): ", lastSemester => {
            const description = `The client is requesting support because they have a disabled account. The last semester the client attended was ${lastSemester}, but has a recent cleared application for this semester.`;
            const fcmNumber = generateFcmNumber(today);
            const resolution = `${fcmNumber}: Verified client information. Re-enabled account. Assisted client in resetting password online. Verified client could sign in. Registered the two sign-in methods on the MyPassword Assistant.`;
            resolve({ description, resolution });
        });
    });
}

function passwordResetEmailCase(today) {
    // Create description and resolution for password reset email case.
    return new Promise(resolve => {
        rl.question("When was the email sent? (MM/DD/YY): ", emailDate => {
            rl.question("At what time was the email sent? (HH:MM): ", emailTime => {
                const description = `Client sent an email on ${emailDate} at ${emailTime} requesting support with a password reset.`;
                const fcmNumber = generateFcmNumber(today);
                const resolution = `${fcmNumber}: Client received password reset auto-response and our contact information.`;
                resolve({ description, resolution });
            });
        });
    });
}

function main() {
    rl.question("Enter today's date (MM/DD/YY): ", todayStr => {
        const today = moment(todayStr, "MM/DD/YY");

        const handleRequest = () => {
            console.log("\nPlease select the type of ticket:");
            console.log("1. Password Reset");
            console.log("2. Disabled Account");
            console.log("3. Compromised Account");
            console.log("4. Password Reset Email");
            console.log("5. Authentication Error");

            rl.question("Enter the number corresponding to the ticket type: ", async ticketType => {
                let description, resolution, internalNotes;

                switch (ticketType) {
                    case '1':
                        const expiryDate = await new Promise(resolve => rl.question("When did the account expire? (MM/DD/YY): ", resolve));
                        ({ description, resolution } = passwordResetCase(expiryDate, today));
                        break;
                    case '2':
                        ({ description, resolution } = await disabledAccountCase(today));
                        break;
                    case '5':
                        ({ description, resolution, internalNotes } = authenticationErrorCase(today));
                        console.log("\nInternal Notes:\n", internalNotes);
                        break;
                    default:
                        console.log("This ticket type is not yet implemented.");
                        handleRequest();
                        return;
                }

                console.log("\nDescription:\n", description);
                console.log("\nResolution:\n", resolution);

                rl.question("Do you want to handle another request? (yes/no): ", answer => {
                    if (answer.toLowerCase() === 'yes') {
                        handleRequest();
                    } else {
                        rl.close();
                    }
                });
            });
        };

        handleRequest();
    });
}

main();
